backend/src/stwo/prover.rs

```diff
@@ -1,32 +1,159 @@
+                LOG_LAST_LAYER_DEGREE_BOUND as u32,
+                FRI_LOG_BLOWUP as u32,
+                FRI_NUM_QUERIES,
+            ),
+        };
+
         Ok(Self {
-            _analyzed,
+            analyzed,
             _fixed,
             _proving_key: None,
             _verifying_key: None,
+            _channel_marker: PhantomData,
+            _backend_marker: PhantomData,
+            _merkle_channel_marker: PhantomData,
         })
     }
+    pub fn prove(&self, witness: &[(String, Vec<F>)]) -> Result<Vec<u8>, String> {
+        let config = get_config();
+        // twiddles are used for FFT, they are computed in a bigger group than the eval domain.
+        // the eval domain is the half coset G_{2n} + <G_{n/2}>
+        // twiddles are computed in half coset G_{4n} + <G_{n}>, double the size of eval doamin.
+        let twiddles = B::precompute_twiddles(
+            CanonicCoset::new(self.analyzed.degree().ilog2() + 1 + FRI_LOG_BLOWUP as u32)
+                .circle_domain()
+                .half_coset,
+        );
+
+        // Setup protocol.
+        let mut prover_channel = <MC as MerkleChannel>::C::default();
+        let commitment_scheme = &mut CommitmentSchemeProver::<B, MC>::new(config, &twiddles);
+
+        let trace = gen_stwo_circuit_trace::<F, B, M31>(witness, self.analyzed.clone());
+
+        let mut tree_builder = commitment_scheme.tree_builder();
+        tree_builder.extend_evals(trace);
+        tree_builder.commit(&mut prover_channel);
+
+        let component = PowdrComponent::new(
+            &mut TraceLocationAllocator::default(),
+            PowdrEval::new(self.analyzed.clone()),
+        );
+
+        let proof = stwo_prover::core::prover::prove::<B, MC>(
+            &[&component],
+            &mut prover_channel,
+            commitment_scheme,
+        )
+        .unwrap();
+
+        Ok(bincode::serialize(&proof).unwrap())
+    }
+
+    pub fn verify(&self, proof: &[u8], _instances: &[F]) -> Result<(), String> {
+        assert!(
+            _instances.is_empty(),
+            "Expected _instances slice to be empty, but it has {} elements.",
+            _instances.len()
+        );
+
+        let config = get_config();
+        let proof: StarkProof<MC::H> =
+            bincode::deserialize(proof).map_err(|e| format!("Failed to deserialize proof: {e}"))?;
+
+        let mut verifier_channel = <MC as MerkleChannel>::C::default();
+        let mut commitment_scheme = CommitmentSchemeVerifier::<MC>::new(config);
+
+        //Constraints that are to be proved
+        let component = PowdrComponent::new(
+            &mut TraceLocationAllocator::default(),
+            PowdrEval::new(self.analyzed.clone()),
+        );
+
+        // Retrieve the expected column sizes in each commitment interaction, from the AIR.
+        let sizes = component.trace_log_degree_bounds();
+        commitment_scheme.commit(proof.commitments[0], &sizes[0], &mut verifier_channel);
```

Comment:
Clarify these 0s

Reply:
add assert to make sure the sizes has length 1 for now. 
Later when constant columns get implemented, witness columns and constant columns will be committed separately and sizes will have length 2
On my way to make constant columns in another PR